From bf1f7a7d8ff9e532932e20edeb4fc54bd34dc8de Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 04 May 2021 17:12:32 +0300
Subject: [PATCH] [SW-40470] gaudi: sec: fixing bug in img verification method

Fixing the logic for choosing the image verification method.
When security descriptor is supported it should be verified.

Change-Id: I2a3551b5677c0a2adf9cd36121d327211235e4aa
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/fit.c b/preboot/fit.c
index 055895f..4624f4b 100644
--- a/preboot/fit.c
+++ b/preboot/fit.c
@@ -779,10 +779,10 @@
 	 * is validated.
 	 */
 	if (preboot_is_image_sig_supported()) {
-		if (fit_verify_security(fit, conf, res))
+		if (preboot_verify_img_sig(fit, fit_size, res))
 			return FIT_LOAD_SEC_FAILED;
 	} else {
-		if (preboot_verify_img_sig(fit, fit_size, res))
+		if (fit_verify_security(fit, conf, res))
 			return FIT_LOAD_SEC_FAILED;
 	}
 
